app/src/main/java/com/duckduckgo/app/bookmarks/ui/bookmarkfolders/BookmarkFoldersActivity.kt

```diff
@@ -85,15 +108,26 @@ class BookmarkFoldersActivity : DuckDuckGoActivity() {
         }
     }
 
+    private fun setNewlyCreatedSelectedFolderResult() {
```

Comment:
This is my proposal to avoid storing those constants in the ViewModel. Just get it from intent.extras since they never change. 

``` 
   private fun setNewlyCreatedSelectedFolderResult() {
        viewModel.newFolderAdded(
                rootFolderName = getString(R.string.bookmarksSectionTitle),
                selectedFolderId = intent.extras?.getLong(KEY_BOOKMARK_FOLDER_ID) ?: 0,
                currentFolder = intent.extras?.getSerializable(KEY_CURRENT_FOLDER) as BookmarkFolder?
        )
    }
```

Reply:
Yes, this will make things simpler. Working on the conflict parts now will update the PR soon along with these implementations.